### PR TITLE
Map to correct shader channels for 1,2 channel input GL textures

### PR DESCRIPTION
**Pull request by bernardkwok**
Fix up channel mapping for 1, 2 channel textures. Swizzle them into the correct place.
Update units tests with wood texture and wood shader examples.
Merge of some ILM changes for image acquire: use std::array and changed to make acquire use a constant identifier + optional fallback color. If and only if latter is specified will a fallback image be created, otherwise acquire will fail.

### CODE REVIEW

File: source/MaterialXRender/OpenGL/GLTextureHandler.cpp [hunk 1]
```diff
@@ -67,17 +67,32 @@ bool GLTextureHandler::acquireImage(std::string& fileName,
         switch (imageDesc.channelCount)
         {
         case 3:
-            internalFormat = GL_RGB;
+        {
+            internalFormat = GL_RGBA;
+            // Map {RGB} to {RGB, 1} at shader access time
+            GLint swizzleMasRGB[] = { GL_RED, GL_GREEN, GL_BLUE, GL_ONE };
+            glTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_RGBA, swizzleMasRGB);
```
Comment (feldstj) [inline source/MaterialXRender/OpenGL/GLTextureHandler.cpp]: I notice that you have swizzleMaskR, and swizzleMasRG, is that a mistake, ie. the missing 'k'?

File: source/MaterialXRender/OpenGL/GLTextureHandler.cpp [hunk 2]
```diff
@@ -67,17 +67,32 @@ bool GLTextureHandler::acquireImage(std::string& fileName,
...
             break;
+        }
         case 2:
-            internalFormat = GL_RG;
+        {
+            internalFormat = GL_RGBA;
             format = GL_RG;
+            // Map {red, green} to {red, alpha} at shader access time
+            GLint swizzleMasRG[] = { GL_RED, GL_RED, GL_RED, GL_GREEN };
+            glTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_RGBA, swizzleMasRG);
```
Comment (feldstj) [inline source/MaterialXRender/OpenGL/GLTextureHandler.cpp]: Same here

File: source/MaterialXRender/Handlers/ImageHandler.h [hunk 3]
```diff
@@ -44,13 +45,13 @@ class ImageSamplingProperties
...
+    int uaddressMode = -1;
     /// Address mode in V
-    int vaddressMode;
+    int vaddressMode = -1;
     /// Filter type
-    int filterType;
+    int filterType = -1;
     /// Default color
-    float defaultColor[4];
+    std::array<float, 4> defaultColor = { { 0.0f, 0.0f, 0.0f, 1.0f } };
```
Comment (bernardkwok) [inline source/MaterialXRender/Handlers/ImageHandler.h]: ILM change.

File: source/MaterialXRender/OpenGL/GLTextureHandler.cpp [hunk 4]
```diff
@@ -53,31 +54,42 @@ bool GLTextureHandler::acquireImage(std::string& fileName,
...
     {
-
         imageDesc.resourceId = MaterialX::GlslProgram::UNDEFINED_OPENGL_RESOURCE_ID;
         glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
         glGenTextures(1, &imageDesc.resourceId);
         glActiveTexture(GL_TEXTURE0 + imageDesc.resourceId);
         glBindTexture(GL_TEXTURE_2D, imageDesc.resourceId);
 
-        GLint internalFormat = GL_RGBA;
+        GLint internalFormat = imageDesc.floatingPoint ? GL_RGBA32F : GL_RGBA;
```
Comment (bernardkwok) [inline source/MaterialXRender/OpenGL/GLTextureHandler.cpp]: ILM fix. Want internal format float if image is float.

File: source/MaterialXRender/OpenGL/GLTextureHandler.cpp [hunk 5]
```diff
@@ -98,26 +110,16 @@ bool GLTextureHandler::acquireImage(std::string& fileName,
         textureLoaded = true;
     }
 
-    if (!textureLoaded)
+    // Create a fallback texture if failed to load
+    if (!textureLoaded && fallbackColor)
```
Comment (bernardkwok) [inline source/MaterialXRender/OpenGL/GLTextureHandler.cpp]: Logic change based on ILM change for a constant filename as input. Now we allow an explicit fallback color which will drive create a solid color image with the same cache key (original file name). This makes it so the client does not need to track another made up cache key name but also now allow for the default color to be passed over if the image does not load (which is more correct then just using black).

**Pull request merged**